include/aws/s3/private/s3_client_impl.h

```diff
@@ -192,6 +192,12 @@ struct aws_s3_client {
      * to meta requests for use. */
     const size_t max_part_size;
 
+    /* The size threshold in bytes for when to use multipart uploads for a AWS_S3_META_REQUEST_TYPE_PUT_OBJECT meta
+     * request. Uploads over this size will automatically use a multipart upload strategy, while uploads smaller than
+     * this threshold will use a single connection to upload the whole object. If not set, `part_size` will be used as
+     * threshold. */
+    const uint64_t multipart_upload_threshold;
```

Comment:
if part sizes are size_t, shouldnt this also be size_t?

Reply:
part size and threshold were long in java v1 tm? and looks like proposal is to have them both be long going forward. its probably better to have the same type for all part related config options, and java can cast if really needed